### PR TITLE
feat:added priority filter and labels page

### PR DESCRIPTION
**Pull request by Hardiegogo**
- added filterbox
- added labels page

image

### CODE REVIEW

**Review by mnath30 — APPROVED** [review]
Code is clean
Liked that error handling is also taken care of
Good naming conventions

File: src/App.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ import NotesListing from "./pages/notes-page/NotesListing";
 import RequiresAuth from "./utils/require-auth/RequiresAuth";
 import ArchiveListing from "./pages/notes-page/ArchiveListing";
```
Comment (VedanthB) [inline src/App.js:10]: you can do a barrel import / export here, basically for all exports / imports. 

makes the code look cleaner

**Review by SanjayJatti — APPROVED** [review]
- liked the folder structure
- all functionality works fine
- code is clean

**Review by aqsaaqeel — APPROVED** [review]
Like the UI and code is easy to understand 👍🏽

File: src/components/notecard-component/NoteCard.jsx [hunk 2]
```diff
@@ -1,86 +1,171 @@
...
+const deletePermanently = (trashNote, dispatchNotes) => {
+  dispatchNotes({ type: "DELETE_NOTE", trashNote: trashNote });
+};
 
+const editHandler = (_id, dispatchNotes, createNote, setCreateNote) => {
+  dispatchNotes({ type: "SET_EDIT_ID", payload: _id });
+  setCreateNote(!createNote);
+};
 
+const changePriority=async(note,
```
Comment (VedanthB) [inline src/components/notecard-component/NoteCard.jsx]: move this into a utils folder

File: src/components/notecard-component/NoteCard.jsx [hunk 3]
```diff
@@ -1,86 +1,171 @@
...
+  dispatchNotes)=>{
+    const encodedToken=localStorage.getItem("token")
+    const {_id, title,color, description, tags, createdAt }=note
+    const changedPriority=[...tags][0]==='low'? 'high' : 'low' ;
+    
+    try {
+      const response=await axios({
```
Comment (VedanthB) [inline src/components/notecard-component/NoteCard.jsx]: move the api calls into a services folder

File: src/components/side-bar-component/Sidebar.jsx [hunk 4]
```diff
@@ -16,7 +16,7 @@ const Sidebar = ({createNoteState:{createNote,setCreateNote}}) => {
       <div className="list-container">
         <ul className="stacked-list">
           <Link to='/notes'><li className={`list-item stacked-list-item ${createNote && 'overlay'}`}><i class='bx bx-home-alt list-icon'></i> Home</li></Link>
```
Comment (VedanthB) [inline src/components/side-bar-component/Sidebar.jsx:18]: you can map over an array and render the list , just a suggestion

File: src/pages/notes-page/NotesListing.jsx [hunk 5]
```diff
@@ -6,16 +6,29 @@ import { useNotes } from "../../context/useNotes";
...
   const { notesState, dispatchNotes } = useNotes();
   const [createNote, setCreateNote] = useState(false);
-
+  const [isFilterBox, setIsFilterBox] = useState(false);
   useEffect(() => {
     loadNotes(dispatchNotes);
+    
   }, []);
 
+  const showNotes=()=>{
```
Comment (VedanthB) [inline src/pages/notes-page/NotesListing.jsx:20]: you can move this into a helpers folder

**Review by VedanthB — APPROVED** [review]
Good work Chirag

**Pull request merged**